test/smoketest.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Artman smoke tests.
+
+It generates both grpc and gapic client libraries for Google APIs in googleapis
+repo (except those being blacklisted), and fail if any generation fails.
+The test currently assumes that artman and googleapis folders are under the same
+parent folder.
+"""
+
+import os
+import fnmatch
+import subprocess
+import sys
+
+ARTMAN_CONFIG_BLACKLIST= [
+  'artman_spanner_admin_instance.yaml',
+  'artman_spanner_admin_database.yaml',
+  'artman_bigtable_admin.yaml'
+]
+
+# Not ready to test Go and Csharp yet.
+SUPPORTED_LANGS = ['python', 'java', 'ruby', 'nodejs', 'php']
+
+def run_smoke_test():
+  artman_config_dir = '../googleapis/gapic/api'
+  for root, dirs, files in os.walk(artman_config_dir):
+    for f in fnmatch.filter(files, 'artman_*.yaml'):
+      if f in ARTMAN_CONFIG_BLACKLIST:
+        # Do not run generation tests for those in the blacklist.
+        continue
+      filename = os.path.join(root, f)
+      content = open(filename).read()
+      for lang in SUPPORTED_LANGS:
+        if '%s:' % lang in content:
+          if generate_grpc_library(filename, lang):
+            sys.exit(
```

Comment:
Initially did this as I don't want to wait until everything is done. But it makes sense to capture everything in one round.